src/pallene/pallenelib.lua

```diff
@@ -43,9 +43,73 @@ return [==[
 #include <stdarg.h>
 #include <locale.h>
 #include <math.h>
+#include <stdbool.h>
 
 #define PALLENE_UNREACHABLE __builtin_unreachable()
 
+/* Part of Pallene Tracer. */
+#define PALLENE_C_FRAMEENTER(L, name)   pt_fn_details_t _details = {                             \
+                                            .fn_name  = name,                                    \
+                                            .mod_name = PALLENE_SOURCE_FILE                      \
+                                        };                                                       \
+                                        pt_frame_t _frame = {                                    \
+                                            .type = PALLENE_TRACER_FRAME_TYPE_C,                 \
+                                            .shared = {                                          \
+                                                .details = &_details                             \
+                                            }                                                    \
+                                        };                                                       \
+                                        pallene_tracer_frameenter(L, &_frame)
+
+#define PALLENE_LUA_FRAMEENTER(L, sig)  pt_frame_t _frame = {                                    \
+                                            .type = PALLENE_TRACER_FRAME_TYPE_LUA,               \
+                                            .shared = {                                          \
+                                                .frame_sig = sig                                 \
+                                            }                                                    \
+                                        };                                                       \
+                                        pallene_tracer_frameenter(L, &_frame)
+
+#define PALLENE_GLOBAL_SETLINE(L, line) pallene_tracer_global_setline(L, line)
```

Comment:
I worry that a GLOBAL_SETLINE function might not be compatible with alternative ways to reference the stack. Perhaps it would be more appropriate to call PALLENE_SETLINE before calling the various "pallene_error" functions? (As opposed to what we do now, where the pallene_error calls global_setline)

Reply:
Removed `PALLENE_GLOBAL_SETLINE` and did workaround by calling `PALLENE_SETLINE` before calling any "pallene_error".